File: 0046-permutations/0046-permutations.py
import itertools # Biblioteca nativa obrigatória para usar o 'permutations'

class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        resultado = []

        # O itertools entrega cada combinação no formato de tupla. Ex: (1, 2, 3)
        for permutacao in itertools.permutations(nums):
            # Precisamos converter a 'permutacao' (tupla) em uma lista padrão [1, 2, 3] 
            resultado.append(list(permutacao))

        return resultado

File: 0046-permutations/0046-permutations.cpp
class Solution {
public:
    vector<vector<int>> permute(vector<int>& nums) {
       vector<vector<int>> resultado;

       sort(nums.begin(), nums.end());
        do {
            
               resultado.push_back(nums);
            
        } while (next_permutation(nums.begin(), nums.end()));
    return resultado;
    }

};